packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -1082,6 +1088,32 @@ class RawGestureDetectorState extends State<RawGestureDetector> {
     _syncAll(widget.gestures);
   }
 
+  @override
+  void didChangeAccessibilityFeatures() {
+    if (!widget.excludeFromSemantics)
+      _maybeUpdateSemanticsActionOverrides();
+  }
+
+  void _maybeUpdateSemanticsActionOverrides() {
```

Comment:
Would it be simpler to always configure the `RenderSemanticsGestureHandler` with all custom actions and then have it figure out based on the valid actions which ones will actually be added to keep the `validActions` logic all in one palce?

Reply:
I am also wondering whether instead of taking CustomSemanticsAction, `RenderSemanticsGestureHandler` should just take a map of action to label.

Reply:
That map could just be passed to the _GestureSemantics and wouldn't have to be updated.